SpatialGDK/Source/SpatialGDK/Private/Utils/InterestFactory.cpp

```diff
@@ -33,6 +34,143 @@ DECLARE_CYCLE_STAT(TEXT("AddUserDefinedQueries"), STAT_InterestFactoryAddUserDef
 
 namespace SpatialGDK
 {
+struct MyQuery
+{
+	TArray<Worker_ComponentId> Components;
+	TArray<Worker_ComponentSetId> ComponentSets;
+	TArray<Worker_EntityId_Key> Entities;
+
+	void DebugOutput(const FString& DiffType) const
+	{
+		// Minimal output
+		if (Components.Find(SpatialConstants::CLIENT_ENDPOINT_COMPONENT_ID) != INDEX_NONE)
+		{
+			UE_LOG(LogTemp, Warning, TEXT("Interest diff: client auth interest"));
+		}
+		else
+		{
+			UE_LOG(LogTemp, Warning, TEXT("Interest diff: client non auth interest"));
+		}
+
+		/*
+		// Verbose output
+		FString ComponentsString;
+		for (const auto& Component : Components)
+		{
+			ComponentsString += FString::Format(TEXT("{0} "), { Component });
+		}
+
+		UE_LOG(LogTemp, Warning, TEXT("Interest diff: components %s"), *ComponentsString);
+
+		FString ComponentSetsString;
+		for (const auto& ComponentSet : ComponentSets)
+		{
+			ComponentSetsString += FString::Format(TEXT("{0} "), { ComponentSet });
+		}
+
+		UE_LOG(LogTemp, Warning, TEXT("Interest diff: component sets %s"), *ComponentSetsString);
+		*/
+
+		FString EntitiesString;
+		for (const auto& EntityId : Entities)
+		{
+			EntitiesString += FString::Format(TEXT("{0} "), { EntityId });
+		}
+
+		UE_LOG(LogTemp, Warning, TEXT("Interest diff: %s %s"), *DiffType, *EntitiesString);
+	}
+};
+
+struct ChangeInterestRequest
```

Comment:
Seems like serialization stuff like this should better sit in the Public/Schema folder?